tests/unit/test_wizard.py

```diff
@@ -383,3 +384,51 @@ def test_stage_exception_handler_other(error_class):
     err = error_class()
     res = stage_error_handler(err, ['stage'], confirm=confirm, prompt=prompt)
     assert res is None
+
+
+@pytest.fixture
+def test_shape():
+    shapes = {
+        "TestShape": {
+            "type": "structure",
+            "members": {
+                "Huge": {"shape": "Long"},
+                "Scale": {"shape": "Double"},
+                "Count": {"shape": "Integer"},
+                "Items": {"shape": "TestList"}
+            }
+        },
+        "TestList": {
+            "type": "list",
+            "member": {
+                "shape": "Float"
+            }
+        },
+        "Long": {"type": "long"},
+        "Float": {"type": "float"},
+        "Double": {"type": "double"},
+        "Integer": {"type": "integer"}
+    }
+    return model.ShapeResolver(shapes).get_shape_by_name('TestShape')
+
+
+def test_param_coercion_numbers(test_shape):
+    # verify coercion will convert strings to numbers according to shape
+    params = {
+        "Count": "5",
+        "Scale": "2.3",
+        "Items": ["5", "3.14"],
+        "Huge": "92233720368547758070"
+    }
+    coerced = ParamCoercion().coerce(params, test_shape)
+    assert isinstance(coerced['Count'], int)
+    assert isinstance(coerced['Scale'], float)
+    assert all(isinstance(item, float) for item in coerced['Items'])
+    assert coerced['Huge'] == 92233720368547758070
+
+
+def test_param_coercion_failure(test_shape):
```

Comment:
Hmm, so how does the error get propogated if an invalid value is passed in like "fifty" from the users prospective?